src/main/java/frc/robot/FieldConstants.java

```diff
@@ -28,20 +29,27 @@ public class FieldConstants {
     public static final Pose2d NOTE_S_2 = new Pose2d(2.43, 5.55, Rotation2d.fromDegrees(0));
     public static final Pose2d NOTE_S_3 = new Pose2d(2.49, 6.81, Rotation2d.fromDegrees(161.94));
 
-    
-    //these are all placeholders and subject to change 
-    public static final Pose2d SHOOT_M_1 = new Pose2d(0, 0 ,Rotation2d.fromDegrees(0));
-    public static final Pose2d SHOOT_M_2 = new Pose2d(0, 0 ,Rotation2d.fromDegrees(0));
+    // these are all placeholders and subject to change
+    public static final Pose2d SHOOT_M_1 = new Pose2d(0, 0, Rotation2d.fromDegrees(0));
+    public static final Pose2d SHOOT_M_2 = new Pose2d(0, 0, Rotation2d.fromDegrees(0));
 
-    public static final Pose2d NOTECHECK_1 = new Pose2d(NOTECHECK_X, 0.78,Rotation2d.fromDegrees(0));
-    public static final Pose2d NOTECHECK_2 = new Pose2d(NOTECHECK_X, 2.44,Rotation2d.fromDegrees(0));
+    public static final Pose2d NOTECHECK_1 = new Pose2d(NOTECHECK_X, 0.78, Rotation2d.fromDegrees(0));
+    public static final Pose2d NOTECHECK_2 = new Pose2d(NOTECHECK_X, 2.44, Rotation2d.fromDegrees(0));
     public static final Pose2d NOTECHECK_3 = new Pose2d(NOTECHECK_X, 5.77, Rotation2d.fromDegrees(0));
     public static final Pose2d NOTECHECK_4 = new Pose2d(NOTECHECK_X, 7.45, Rotation2d.fromDegrees(0));
 
-
-
-    public static final Pose2d START_1 = new Pose2d(0, 0 ,Rotation2d.fromDegrees(0));
+    public static final Pose2d START_1 = new Pose2d(0, 0, Rotation2d.fromDegrees(0));
     public static final Pose2d START_2 = new Pose2d(1.3, 5.53, Rotation2d.fromDegrees(0));
-    public static final Pose2d START_3 = new Pose2d(0, 0 ,Rotation2d.fromDegrees(0));
-
+    public static final Pose2d START_3 = new Pose2d(0, 0, Rotation2d.fromDegrees(0));
+
+    public static Pose2d flipPose(Pose2d pose) {
+        // flip pose when red (have to do this because PathPlanner flips a path only)
+        if (DriverStation.getAlliance().get() == Alliance.Red){
```

Comment:
What about if it is not present??

Reply:
then it would just return the pose from the parameter. Or should I write it in an else to make it clear?

Reply:
I would just return Blue position if getAlliance is not present; that can happen before the robot connects to the FMS.